### PR TITLE
chore: update `mocha` version to latest version that support `IE11`

### PR DESCRIPTION
**Pull request by wsuwt**
## Description

Our test helper loads old version of mocha. It's got a dependency with security vulnerability in it. We should look to upgrade to avoid this and also modernise our development tech stack as much as possible.

- Update `mocha` to version 9, the latest version that support IE11
- Removing `esm` package and `--require esm` in mocha cli. ESM is natively supported by Node 16 and above. 

Below changes are in sync with `v7`

- Replacing `require` with `import`
- Changing test files extension to `.test.js` for consistency.
- Making `@refinitiv-ui/phrasebook` type module to support `mocha@9` which is an ESM package.

## Type of change

Please delete options that are not relevant.

- [x] Refactor (improves code without changing its functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/phrasebook/__test__/locale.test.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ const assembleImports = () => {
       let componentsList = getDirList(`./${DEFAULT_LANG_DIR}/${lang}`);
       if(componentsList) {
         for (const component of componentsList) {
-          let imported = require(`../${DEFAULT_LANG_DIR}/${lang}/${component}`);
+          let imported = import(`../${DEFAULT_LANG_DIR}/${lang}/${component}`);
```
Comment (wattachai-lseg) [inline packages/phrasebook/__test__/locale.test.js]: dynamic import returns `Promise`. Shall we `await` here?

File: packages/test-helpers/package.json [hunk 2]
```diff
@@ -28,8 +28,8 @@
     "access": "public"
   },
   "dependencies": {
-    "@open-wc/testing": "^3.0.0-next.5",
-    "@types/mocha": "^5.2.7",
-    "mocha": "^6.2.3"
+    "@open-wc/testing": "^3.2.0",
+    "@types/mocha": "^9.1.1",
```
Comment (wattachai-lseg) [inline packages/test-helpers/package.json:32]: `@types/mocha` should be devDep instead?

**Review by wattachai-lseg — CHANGES_REQUESTED** [review]
2 things plus a merge conflict.

File: packages/test-helpers/package.json (hunk 2 above)
Comment (wsuwt) [reply packages/test-helpers/package.json:32]: I think it should remain in dependencies. We ship mocha and type as a dependencies to users (mainly us).

File: packages/phrasebook/__test__/locale.test.js (hunk 1 above)
Comment (wsuwt) [reply packages/phrasebook/__test__/locale.test.js]: I'm not sure if this is being resolved on `done()` after this function is called or not but this is how `v7` is currently. Let me do a bit more investigation on this.

Comment (wattachai-lseg) [reply packages/phrasebook/__test__/locale.test.js]: indeed, `v7` uses `await`. I still think this is the way. We're better off without `done()` 😄

Comment (wsuwt) [reply packages/phrasebook/__test__/locale.test.js]: Updated. I'll create another PR for `v7` to use `async/await` as well.

**Review by wattachai-lseg — APPROVED** [review]

**Pull request merged**